src/cascadia/TerminalSettingsEditor/MainPage.cpp

```diff
@@ -64,69 +64,95 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+        if (const auto selectedItem{ SettingsNav().SelectedItem() })
         {
-            if (const auto navViewItem{ menuItems.GetAt(i).try_as<MUX::Controls::NavigationViewItem>() })
+            if (const auto navViewItem{ selectedItem.try_as<MUX::Controls::NavigationViewItem>() })
+            {
+                selectedItemTag = navViewItem.Tag();
+            }
+            menuItems.IndexOf(selectedItem, *selectedItemIndex);
+        }
+
+        // remove all profile-related NavViewItems by populating a std::vector
+        // with the ones we want to keep.
+        // NOTE: menuItems.Remove() causes an out-of-bounds crash. Using ReplaceAll()
+        //       gets around this crash.
+        std::vector<IInspectable> menuItemsSTL;
+        for (uint32_t i = 0; i < menuItems.Size(); ++i)
+        {
+            const auto item{ menuItems.GetAt(i) };
+            if (const auto navViewItem{ item.try_as<MUX::Controls::NavigationViewItem>() })
             {
                 if (const auto tag{ navViewItem.Tag() })
                 {
                     if (tag.try_as<Editor::ProfileViewModel>())
                     {
-                        // hide NavViewItem pointing to a Profile
-                        navViewItem.Visibility(Visibility::Collapsed);
+                        // don't add NavViewItem pointing to a Profile
+                        continue;
                     }
                     else if (const auto stringTag{ tag.try_as<hstring>() })
                     {
                         if (stringTag == addProfileTag)
                         {
-                            // hide NavViewItem pointing to "Add Profile"
-                            navViewItem.Visibility(Visibility::Collapsed);
+                            // don't add the "Add Profile" item
+                            continue;
                         }
                     }
                 }
             }
+            menuItemsSTL.push_back(item);
         }
-        _InitializeProfilesList();
+        menuItems.ReplaceAll(menuItemsSTL);
 
-        _RefreshCurrentPage();
-    }
+        // Repopulate profile-related menu items
+        _InitializeProfilesList();
 
-    void MainPage::_RefreshCurrentPage()
-    {
-        auto navigationMenu{ SettingsNav() };
-        if (const auto selectedItem{ navigationMenu.SelectedItem() })
+        // now that the menuItems are repopulated,
+        // refresh the current page using the SelectedItem data we collected before the refresh
+        if (selectedItemTag)
         {
-            if (const auto tag{ selectedItem.as<MUX::Controls::NavigationViewItem>().Tag() })
+            if (const auto tag{ selectedItemTag->try_as<hstring>() })
             {
-                if (const auto oldProfile{ tag.try_as<Editor::ProfileViewModel>() })
+                // SelectedItem is a menu item we didn't remove
+                SettingsNav().SelectedItem(menuItems.GetAt(*selectedItemIndex));

```

Comment:
Seems like we should just be able to find the one with a matching tag. No need to keep the index -- what does it do for us?

Reply:
In fact, if the order _does_ ever change, we will navigate to one tag but select the index for a _different_ tag.